thunder/core/transform_common.py

```diff
@@ -476,26 +516,36 @@ def is_functionalizable(bsym: BoundSymbol) -> bool:
         bsyms.append(new_bsym)
 
     intermediate_trace = from_trace(computation_trace)
-    intermediate_trace.bound_symbols = bsyms[:]
+    intermediate_trace.bound_symbols = bsyms
     intermediate_trace.set_provenance(TraceProvenance("Intermediate trace of `functionalize_inplace_ops`"))
-    del bsyms
```

Comment:
Maybe we keep the deletion? That would make it clear that it's now owned by `intermediate_trace`.